### ISSUE TITLE
Testbench not initialising instruction memory correctly when Jump/branch instructions are used

### ISSUE THREAD

**Issue by guptasanchi** [SimpleCPU/SimpleCPU#7]
I was trying to test the newly added support for Jump and branch instruction on the RTL but saw that the model and RTL diverge to different paths on taking a branch or a jump. Even though RTL calculates the correct branch address but the testbench hasn't initialised the locations in the instruction memory correctly. 

Please update the testbench to use the PC information (available from hex_gen tool) to initialise the instruction memory correctly.

**Comment by raulbehl** [comment]
This looks like an expected bug only. The testbench currently initialises the instruction memory sequentially i.e. every other instruction is placed at previous PC + 4 location. I would update the testbench code to use the PC information available from the hex gen tool. 

Thanks

**Comment by raulbehl** [comment]
I have checked in the changes in the testbench to use the pc_values_hex file for initialising the instruction memory. The testbench and the RTL work well for the Jump instructions. Though there seems to be some issue with the way hex_gen is generating Jump instructions. What I noticed was that the hex_gen takes care of the jump address to be within the TEXT area but the subsequent instructions do not follow this i.e. in one of the test I saw the jump instruction update the PC to 0xFFC (which is within the TEXT area - 0x0000-0x1000) but then the subsequent instruction were generated at PC values of 0x1000, 0x1004, etc which is incorrect. As those values are not within the text region. 
There should be a check placed whenever the PC is updated. I will open a new bug for this.
